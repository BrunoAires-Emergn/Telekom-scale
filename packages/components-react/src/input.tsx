import * as React from 'react'

export const Input: React.SFC<{}> = () => <input type="text" placeholder="Some text" />
