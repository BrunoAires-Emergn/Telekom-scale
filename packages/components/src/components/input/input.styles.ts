import { JssStyle } from 'jss';

export const styles: JssStyle = {
  input: {
    display: 'flex',
    width: '100%',
    border: '1px solid #dfdfdf',
    borderRadius: '4px',
    padding: '0.5rem',
    fontSize: '16px',
    fontFamily: 'inherit',
    boxSizing: 'border-box',
    '&:not(.tabbing):focus': {
      outline: 0,
    },
  },
};
