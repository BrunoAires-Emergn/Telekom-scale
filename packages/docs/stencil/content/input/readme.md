# t-input-text



<!-- Auto Generated Below -->


## Properties

| Property       | Attribute       | Description                         | Type                                                                        | Default     |
| -------------- | --------------- | ----------------------------------- | --------------------------------------------------------------------------- | ----------- |
| `customClass`  | `custom-class`  | (optional) Input text class         | `string`                                                                    | `''`        |
| `errorMessage` | `error-message` | (optional) Input text error message | `string`                                                                    | `undefined` |
| `name`         | `name`          | (optional) Input name               | `string`                                                                    | `''`        |
| `styles`       | --              | (optional) Injected jss styles      | `StyleSheet<string \| number \| symbol>`                                    | `undefined` |
| `type`         | `type`          | (optional) Input type               | `"email" \| "hidden" \| "number" \| "password" \| "tel" \| "text" \| "url"` | `'text'`    |
| `value`        | `value`         | (optional) Input text value         | `string`                                                                    | `undefined` |


## Events

| Event     | Description                         | Type                  |
| --------- | ----------------------------------- | --------------------- |
| `changed` | (optional) Input text event changed | `CustomEvent<string>` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
