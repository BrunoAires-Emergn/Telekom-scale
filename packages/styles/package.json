{
    "name": "@telements/styles",
    "version": "0.0.1",
    "main": "dist/index.js",
    "scripts": {
        "build": "tsc -b",
        "postbuild": "node ./dist/export-css.js",
        "setup": "yarn"
    },
    "author": "@eeegor <egor.kirpichev@telekom.de>",
    "license": "MIT",
    "devDependencies": {
        "typescript": "^3.4.5"
    },
    "dependencies": {

    }
}
