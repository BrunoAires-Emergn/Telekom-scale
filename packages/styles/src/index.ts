import { button } from './button';
import { alert } from './alert';
import { card } from './card';

const theme = {
	colors: {
		white: '#fff',
		black: '#000',
		blue: '#00f',
		red: '#f00',
	}
}

export {
	theme,
	alert,
	button,
	card,
}
