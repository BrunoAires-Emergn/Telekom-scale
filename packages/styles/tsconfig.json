{
	"extends": "../tsconfig.base",
	"compilerOptions": {
		"rootDir": "./src",
		"outDir": "./dist",
		"types": [
			"node",
			"jest"
		]
	}
}
