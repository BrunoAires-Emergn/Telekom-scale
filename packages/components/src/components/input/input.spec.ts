import { newSpecPage } from '@stencil/core/testing';
import { Input } from './input';
import { styles } from './input.styles';
import jss from 'jss';

describe('Input', () => {
  let element;
  let stylesheet;
  beforeEach(async () => {
    element = new Input();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Input],
      html: `<t-input></t-input>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  // it('should handle a custom css class', () => {
  //   element.customClass = 'custom';
  //   expect(element.getCssClassMap()).toContain('custom');
  // });

  // it('should handle theme css class', () => {
  //   element.theme = 'default';
  //   expect(element.getCssClassMap()).toContain('input--theme-default');
  // });

  // it('should have a default css class', () => {
  //   expect(element.getCssClassMap()).toContain('input');
  // });
});
