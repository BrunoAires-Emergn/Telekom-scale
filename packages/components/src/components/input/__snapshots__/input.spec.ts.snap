// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<t-input>
  <mock:shadow-root>
    <style>
      .input-0-0-2 {
  width: 100%;
  border: 1px solid #dfdfdf;
  display: flex;
  padding: 0.5rem;
  font-size: 16px;
  box-sizing: border-box;
  font-family: inherit;
  border-radius: 4px;
}
.input-0-0-2:not(.tabbing):focus {
  outline: 0;
}
    </style>
    <input class="input-0-0-2" type="text">
  </mock:shadow-root>
</t-input>
`;
